src/core/init/manifest_update_scheduler.ts

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  EMPTY,
+  merge as observableMerge,
+  Observable,
+  ReplaySubject,
+  timer as observableTimer,
+} from "rxjs";
+import {
+  finalize,
+  ignoreElements,
+  mergeMap,
+  startWith,
+  switchMap,
+  take,
+  tap,
+} from "rxjs/operators";
+import log from "../../log";
+import Manifest from "../../manifest";
+import isNonEmptyString from "../../utils/is_non_empty_string";
+import { IFetchManifestResult } from "../pipelines";
+
+export type IManifestFetcher =
+    (manifestURL? : string, externalClockOffset?: number) =>
+      Observable<IFetchManifestResult>;
+
+/**
+ * Refresh the Manifest at the right time.
+ * @param {Object} initialManifest
+ * @param {Observable} scheduleRefresh$
+ * @param {Function} fetchManifest
+ * @returns {Observable}
+ */
+export default function manifestUpdateScheduler(
+  initialManifest : { manifest : Manifest;
+                      sendingTime? : number; },
+  scheduleRefresh$ : Observable<number>,
+  fetchManifest : IManifestFetcher,
+  minimumManifestUpdateInterval : number
+) : Observable<never> {
+  // Emit each time the manifest is refreshed.
+  const manifestRefreshed$ = new ReplaySubject<{ manifest : Manifest;
```

Comment:
Can't you use a recursive function instead of creating a Subject ?

Reply:
Too hard